### ISSUE TITLE
Bug: `no-floating-promises` should not fire on `Promise.resolve()`

### ISSUE THREAD

**Issue by benmccann** [typescript-eslint/typescript-eslint#9192]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=FABQTg9gtglgzgUwHRgXCAbAbggFASgG5gg&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiRwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Promise.resolve();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-floating-promises": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The rule should not trigger because it's impossible for `Promise.resolve()` to fail. Thus, there is no need for a catch. The rule should only trigger when a promise is passed to `Promise.resolve`

### Actual Result

> Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator. 2:1 - 2:19
> > Add void operator to ignore.

**Comment by Josh-Cena** [comment]
In what situation would you end up with code like this?

I think in general it's impossible for us to detect which promises can never fail, and we have several solutions:

- The work-in-progress https://github.com/typescript-eslint/typescript-eslint/pull/9186
- Using `void`, which is a very lightweight indicator that you wouldn't care

I see that in https://github.com/sveltejs/svelte/pull/11842, your code actually looks like:

```js
Promise.resolve().then(() => {
  // ...
});
```

But then the `then` handler failing would still result in a rejected promise which you need to handle.

**Comment by benmccann** [comment]
> But then the then handler failing would still result in a rejected promise which you need to handle.

Good point
